transport/http/dial.go

```diff
@@ -130,7 +133,7 @@ func newClientNewAuth(ctx context.Context, base http.RoundTripper, ds *internal.
 			DefaultScopes:           ds.DefaultScopes,
 			SkipValidation:          skipValidation,
 		},
-		UniverseDomain: ds.UniverseDomain,
+		UniverseDomain: ds.GetUniverseDomain(),
```

Comment:
Do we need to do this in the new library as well for those that use it directly?

Reply:
Yes, `DialSettings.UniverseDomain` usages should be replaced anywhere they occur with `DialSettings.GetUniverseDomain()` in order to support the env var.

Reply:
The new lib does not have dialsettings, though. But it does allow someone to explicitly set this value. But if they don't, should we honor this there? If we did honor this in the new library and had the current code here I think the bug would be fixed for both, right?

Reply:
I will update this in `grpc/dial.go` in this PR, thank you.